streams-bootstrap-core/src/main/java/com/bakdata/kafka/util/ConsumerGroupClient.java

```diff
@@ -74,7 +75,12 @@ public void deleteConsumerGroup(final String groupName) {
         } catch (final InterruptedException ex) {
             Thread.currentThread().interrupt();
             throw new KafkaAdminException("Failed to delete consumer group " + groupName, ex);
-        } catch (final ExecutionException | TimeoutException ex) {
+        } catch (final ExecutionException ex) {
+            if (ex.getCause() instanceof RuntimeException) {
```

Comment:
See comment below

Reply:
So all those checks are only required for `deleteGroupIfExists()`?

Reply:
I needed access to the cause of the execution exception. And for consistency I throw the cause for all execution exceptions

Reply:
Maybe a retry in `deleteGroupIfExists` would be simpler? That keeps all the logic in there

Reply:
I mean there is no reason to retry if we already know by the Exception type that the group does not exist. If another exception is thrown, the job would crash and at least Kubernetes would restart the pod

Reply:
I feel like adding code for retry would make this much more complex without any benefit